### ISSUE TITLE
Support pasting multiple isrcs.

### ISSUE THREAD

**Issue by kepstin** [kepstin/magicisrc#1]
There should be the ability to paste text containing a list of multiple ISRCs, so they can be assigned sequentially to tracks. This saves a lot of effort individually copy/pasting ISRCs into separate fields.

An enhancement would be to parse certain common formats, e.g. cue files, to assign isrcs to tracks smartly.

**Comment by jesus2099** [comment]
The attention points are
- to mark duplicates (especially consecutives but not only), can be useful even without the mass paste
- to understand when a track has no ISRC and skip it

:)

**Comment by kepstin** [comment]
Duplicate marking is actually already present (along with a bunch of other sanity checks - e.g. for all-zero codes, and codes with unassigned country codes)

**Comment by kepstin** [comment]
Prototype UI. I'm currently working on the actual parsing backend code.

![Screenshot_2019-11-29 kepstin%u2019s MagicISRC (Beta)](https://user-images.githubusercontent.com/38788/69977705-46797a80-14f9-11ea-948b-6250cafe5350.png)

**Comment by jesus2099** [comment]
Hello Kepstin.
Actually I did my own dirty script to allow mass paste but I don’t publicise it, as I noticed I often have to loose hours fixing some badly mass submitted ISRC where some editor did not check the alignment of tracks, or missing tracks, etc.

**Comment by kepstin** [comment]
I've done an initial pass of multiple ISRC paste which is live now on https://magicisrc-beta.kepstin.ca/ - I'd appreciate any feedback.

This current implementation is a simple spread-sheet like thing where any time there's a line break in the pasted input, it will move to the next field. So this is useful for copy/pasting out of spreadsheets or simple text files. It makes no attempt to only find ISRCs in the pasted text.

I would still like to add some sort of parsing support at some point to handle more complex cases.

**Comment by ToadKing** [comment]
I use @jesus2099's script for pasting in CUE sheets with ISRC fields, so the beta page doesn't quite work for my case.

An example cue file: [sample.cue.txt](https://github.com/kepstin/magicisrc/files/6970803/sample.cue.txt)

**Comment by kepstin** [comment]
Right. For cue sheets, I really want to get a proper parser in place, so it can do things like skip tracks without isrcs.

FWIW: I'm not going to deploy the beta paste feature on https://magicisrc.kepstin.ca/ until I've gotten a better parsing solution in place or jesus2099's userscript has been updated to be compatible.

**Comment by jesus2099** [comment]
Thanks @kepstin, it looks great to manage tracks with no ISRC (skip track).
Still missing the ISRC match in each line: https://github.com/jesus2099/konami-command/issues/611

**Comment by kepstin** [comment]
I've added support for matching/extracting the IRSC in pasted text, and released the multiple paste feature in [v20240922.0](https://github.com/kepstin/magicisrc/releases/tag/v20240922.0).

I would still like to add a proper parser for things like CUE sheets or ISRC logs at some point…

**Comment by jesus2099** [comment]
Thanks @kepstin!
I see there is this commit from you: 72a09b7929d9264a6ab85c6d69ab9f527b74b9b9
It looks a lot more compact than my old https://github.com/kepstin/magicisrc/pull/2/files

I am eager to test it, next time!

**Comment by ToadKing** [comment]
Seems to work good here. One suggestion I would make: If a line doesn't contain any text matching an ISRC, it should try to skip to a line that does before advancing to the next track to paste into. That would better match the old userscript and work well enough for CUE files that don't have any tracks missing ISRCs and should be good enough until proper CUE parsing is implemented IMO.

**Comment by jesus2099** [comment]
So magicisrc is currently in one-ISRC-per-line mode ?
And skips one MB track for each row without ISRC?

Sorry I had no time to test, yet. 

Interesting for minc source!

Maybe we could try to **detect** the source and **switch between two modes**: this **row mode**, or, MASS ISRC **bulk mode**?

It could be difficult, but cool.

Row mode is expecting no less and no more than 1 ISRC on each row, representing a track, and it will skip MB tracks for each source row without ISRC.
Bulk mode means ISRC can be on same row, separate rows, anything, it just takes them in order.

**Comment by jesus2099** [comment]
Ach, How can I use cuesheets to mass paste ISRC now?
My MASS ISRC userscript is short circuited, I am trying to make it run first... 🤔
